libs/settings/NRF52Flash.cpp

```diff
@@ -56,21 +98,40 @@ int ZFlash::writeBytes(uintptr_t dst, const void *src, uint32_t len) {
 
     len >>= 2;
 
-    NRF_NVMC->CONFIG = NVMC_CONFIG_WEN_Wen;
-    waitForLast();
+#ifdef SOFTDEVICE_PRESENT
+    if (ble_running())
+    {
+        flash_op_complete = false;
 
-    while (len-- > 0) {
-        uint32_t v = *sp++;
-        if (v != 0xffffffff) {
-            *dp++ = v;
-            waitForLast();
-        } else {
-            dp++;
+        while(1)
+        {
+            if (sd_flash_write((uint32_t *)dp, (uint32_t *)sp, len) == NRF_SUCCESS)
+                break;
+
+            system_timer_wait_ms(10);
```

Comment:
I think this may be too slow for anything useful. Maybe you can spin on flash_op_complete?

Reply:
If the previous call does not go through then flash_op_complete will never trigger as the async callback will never take place. This is the same code that is used in microbit codal now!

Reply:
so you're saying the wait_ms is usually not hit?

Reply:
From my observations during my primitive testing, I have not yet hit this condition. I have only ever gotten NRF_SUCCESS back from these calls.